lib/charms/alertmanager_k8s/v0/alertmanager_dispatch.py

```diff
@@ -70,10 +69,21 @@ class RelationManagerBase(Object):
         name (str): consumer's relation name
     """
 
-    def __init__(self, charm: CharmBase, relation_name):
+    def __init__(self, charm: CharmBase, relation_name: str, relation_role: RelationRole):
         super().__init__(charm, relation_name)
+        self.charm = charm
+        self._validate_relation(relation_name, relation_role)
         self.name = relation_name
 
+    def _validate_relation(self, relation_name: str, relation_role: RelationRole):
+        try:
+            if self.charm.meta.relations[relation_name].role != relation_role:
+                raise ValueError(
+                    f"Invalid role for relation: {relation_name}; must be '{relation_role}'"
```

Comment:
This is not informative as an error message: it should read something like "Relation XYZ in the charm's metadata.yaml should be {provided|required} to be managed by this library, but instead it is {required|provided}"